WIP: Adding OpenTelemetry

kjeffc opened the pull request:
Builds but search is broken currently

ericmaino committed:
PACKAGE: Add missing dependency add build

ericmaino committed:
FIX: Stop using DateTime and move to DateTimeOffset

This change only targets the Item provider at th present moment, howver a bigger change is likely needed to eradicate the usage of DateTime in the codebase

ericmaino committed:
REFACTOR: List<T> dependency to IList<T>

ericmaino committed:
FIX: Update IItemProvider to be an async implemenation

ericmaino committed:
TEST: Remove outdated Arriba.testsettings

ericmaino committed:
FIX: Possible infinite loop?

ericmaino committed:
PLATFORM: Updating most projects to .NET Core

ericmaino committed:
FIX: Arriba.Range conflict System.Range

ericmaino committed:
BREAKING: Dependency on Microsoft.Win32.Registry

ericmaino committed:
BREAKING: Dependency on WindowsIdentity

ericmaino committed:
DUPLICATION: TfsWorkItemCrawler to WorkItemCrawler

- Duplicate the existing project
- Next steps
   - Remove hard TFS dependencies
   - Migrate to .NET Core

ericmaino committed:
PLATFORM: Remove non-Core supported features

Commit:
BUILD: Adding dotnetcore Github Build Action

ericmaino committed:
REFACTOR: Promote DictionaryValueBag

ericmaino committed:
REFACTOR: Introduce ArribaHttpContextRequest

- Add support for both Owin and HttpContext in order to enable aspnet
  controller support moving forward
- Note this implementation is not fully flushed out as things like
  an identity will need to be added once OAuth support exists

ericmaino committed:
FEATURE: Add support aspnet controllers

ericmaino committed:
FIX: Disable early disposal of stream

- Note this appears to be causin and IO issue when running under kestrel

ericmaino committed:
DEBUG: Update IIS to run on 42784

Commit:
BUILD: Address warnings and other .NET Core build issues

Commit:
TEST: Ignore failing tests. Enable build/test gate

ericmaino committed:
FEATURE: Naive Core Azure DevOps Crawler

ericmaino committed:
REFACTOR: Move to src to better support docker builds

meeteric-mob committed:
FEATURE: Add docker support for local debugging

meeteric-mob committed:
REFACTOR: Extract Arriba.Configuration

ViniciusSouza committed:
FIX: Adding script to execute crawler in infinite loop

If the process exits the Azure Web App using compose fails

ViniciusSouza committed:
FEATURE: Adding Enable Authentication flag

ericmaino committed:
DEBUG: Adding support for appsettings.local.json

ericmaino committed:
FIX: Add support for Tags as a string set

ericmaino committed:
PACKAGE: Updating Newtonsoft.json v9.0.1 to v10.0.1

ericmaino committed:
FEATURE: Adding OAuth authentication support

ericmaino committed:
FEATURE: Adding CORS support

ericmaino committed:
FEATURE: Unify configuration processing

ericmaino committed:
DEBUG: Improve debugging messages on .NET Core

ericmaino committed:
FIX: Path related bugs when running on linux

ViniciusSouza committed:
FIX: Adding Authorization header to XHR method

ViniciusSouza committed:
FIX: Using the frontend url as the service base

When using proxy as NGINX the /data can be used

ViniciusSouza committed:
FIX: Adding mime.types for NGINX

SVG was not properly rendered

ViniciusSouza committed:
FEATURE: Adding docker-compose setup document

ViniciusSouza committed:
FIX: Adding server configuration to .docker\Server

Following the pattern of the other services

ericmaino committed:
FIX: Do not delete existing settings

- The image should honor what is on disk and should not work around accidental
  configuration isseus by deleting them

ericmaino committed:
CONFIG: Use default for the default config

- Removes reference from cseng in the docker file and disk

ericmaino committed:
FIX: Ensure environment variables are honored for additional sources

ericmaino committed:
FIX: Increase column suggestion to show top 25

ericmaino committed:
FIX: Deny create access until authorization bug is fixed

ViniciusSouza committed:
TESTING: Add tests to validate the new service

ViniciusSouza committed:
FIX: Doesn't throw exception when key is not found

ViniciusSouza committed:
FIX: Removed unused variable

ViniciusSouza committed:
FIX: GetTableInformationForUser checks for Reader scope

Added a condition to verify if the user has at least reader scope

ViniciusSouza committed:
FEATURE: Expanding IArribaManagementService

ViniciusSouza committed:
REFACTOR: Implementing IArribaManagementService

ViniciusSouza committed:
FEATURE: Add guard type ParamChecker

ViniciusSouza committed:
REFACTOR: Remove handle chain from ArribaTableRoutesApplication

As the service already check permission and validate the request
I removed the handle chain methods from the routes specified on ArribaManagement

On branch visouza/wip-server-refactor
Changes to be committed:
	modified:   src/Arriba/Arriba.Communication/Server/Application/ArribaManagement.cs

ViniciusSouza committed:
FEATURE: Add type ArribaManagementService

The go of this change is to have a separate type for the service
other than use the ArribaManagement
that was used before to avoid code duplication as the refactor was being done.

ViniciusSouza committed:
REFACTOR: Decouple Authorization from ArribaApplication

ViniciusSouza committed:
REFACTOR: Support for Composition

ArribaManagement uses System.Composition need to perform some
changes to allow the ArribaManagementService usage

ViniciusSouza committed:
REFACTOR: Rename ArribaManagement -> ArribaTableRoutesApplication

ViniciusSouza committed:
REFACTOR: Decouple ArribaManagement from ArribaTableRoutesApplication

ericmaino committed:
Merge pull request #11 from ViniciusSouza/visouza/main-ArribaManagementRefactor-clean-history

REFACTOR: Arriba Management

ericmaino committed:
CONFIG: Clean up docker commands and config

- Removes some unnecessary commands
- Ensures that default config values are not valid to ensure proper overriding happens

ViniciusSouza committed:
PIPELINE: Update linux version to ubuntu-20.04

ViniciusSouza committed:
TESTING: Enabling disabled tests

ViniciusSouza committed:
FIX: Unit test Value_Basic

The casting to double in dotnet core has more precision (three more digits
was added).

The same happened for others casting what make the test failing when
converting from .Net Standard to dotnet core

ViniciusSouza committed:
FIX: Unit test ParseQuotedNewLine

The test was using a Windows new line (carriage return) code '/r/n'
to digest and compare the values.

Instead of using the carriage return code I included the usage of
Environment.NewLine that returns the correct value depending on the
OS (Unix Based/Windows Based).

ericmaino committed:
DOCKER: Update to default to published images

ViniciusSouza committed:
FIX: Unit test DistributionQuery_Rounding

Specifying the expected datetime format used for comparison
Not getting impacted by the system cultural definition.

ericmaino committed:
FEATURE: Enable disbling crawler for base image

ericmaino committed:
FIX: Add offline_access to the default scopes

ericmaino committed:
Merge pull request #14 from ViniciusSouza/visouza/main-tests-fix

TESTING: Enable disabled tests and fix them

ViniciusSouza committed:
PROJECT: Change default namespace to Arriba

ViniciusSouza committed:
FEATURE: Disabling authentication for developer loop

ViniciusSouza committed:
REFACTOR: Remove unused property

ViniciusSouza committed:
REFACTOR: Decouple Start from Program

ericmaino committed:
REFACTOR: Cleanup namspaces on Arriba.WorkItemCrawler

ericmaino committed:
REFACTOR: Promote Arriba.Query into Arriba

ericmaino committed:
TEST: Address test to normalize new line inconsistencies

ericmaino committed:
TEST: Fix a bug where tests fail when in src folder

ericmaino committed:
PROJECT: Adding Arriba.Adapters.Netwonsoft

ViniciusSouza committed:
REFACTOR: Removing unused code

ViniciusSouza committed:
REFACTOR: Start.Configure Use IHostEnvironment

ViniciusSouza committed:
REFACTOR: Move Configuration to Configuration folder

ViniciusSouza committed:
RENAME: ArribaContoller to ArribaController

ViniciusSouza committed:
FEATURE: Add Exception Middleware

ViniciusSouza committed:
FEATURE: Add ArribaResultFilter

ArribaRoutedApplication uses ArribaResponse that envelopes the service result.
This filter will envelop the result for ArribaController

ViniciusSouza committed:
FEATURE: Add new route at nginx for ArribaController

At this point the two implementation will run in parallel.
This way it is possible to check any odd behavior that may occur.

ViniciusSouza committed:
FEATURE: Implement ArribaController

ViniciusSouza committed:
REFACTOR: Use route datav2 at the backend url

Nginx has two routes data and datav2:
**data** uses the ApplicationRoutedTable services
**datav2** users the ArribaController services

The both routes will be running in parallel for some time.

ViniciusSouza committed:
REFACTOR: Using Arriba as namespace

Removing Server from the namespace.

ericmaino committed:
REFACTOR: Move to unified Arriba namespace

ericmaino committed:
REFACTOR: Move all converters into Newtonsoft adapter

ericmaino committed:
REFACTOR: Unifi serialization configuration process

ericmaino committed:
REFACTOR: Move Json reader/writers to adapter assembly

ericmaino committed:
REFACTOR: Remove direct dependency on JsonConvert

ericmaino committed:
REFACTOR: Remove dependency on JsonSerializationException

ericmaino committed:
REFACTOR: Remove direct dependency on StringEnumConverter

ericmaino committed:
TEST: Add test to validate SecureSet serialization

ericmaino committed:
REFACTOR: Remove Newtonsoft JsonArray dependency

ericmaino committed:
REFACTOR: Move SecureSet into separate file

ericmaino committed:
REFACTOR: Renami Arriba.Configuration - Arriba.Adapter.Configuration

ericmaino committed:
REFACTOR: Combine Client & Communication into Composition

- Combining code into a single assembly, most of which will be promoted into Arriba in the near future

ericmaino committed:
REFACTOR: Promote types to Arriba requiring access changes

ericmaino committed:
REFACTOR: Promote types into Arriba project

ericmaino committed:
REFACTOR: Remove direct dependency on MemoryCache

- Introduces IObjectCache and IObjectCacheFactory
- Moves MemoryCache dependency into a very specific implementaiton that can next be moved into an adapter.

ericmaino committed:
REFACTOR: Reducing direct depdendencies on composition libraries

ericmaino committed:
REFACTOR: Promoting classes into Arriba project

ericmaino committed:
TEST: Adding a test to verify healthy composition

ericmaino committed:
REFACTOR: Decouple implementation from composition

ericmaino committed:
PROJECT: Adding Arriba.Adapter.Cache

ericmaino committed:
REFACTOR: Moving MemoryCacheFactory into Arriba.Adapter.Cache

ericmaino committed:
PROJECT: Adding Arriba.Adaapter.Telemetry

ericmaino committed:
REFACTOR: Move DataFlow dependenices into Telemetry adapter

ericmaino committed:
REFACTOR: Push ArribaConvert up into Arriba library

- Note I'm still not a fan of this change, but the plumbing required to address it is more then I'd like take on at the present moment.

ericmaino committed:
REFACTOR: Removing direct Newtonsoft dependency

ericmaino committed:
FIX: Address is issue with composition initialization

ericmaino committed:
PROJECT: New Arriba.Adapter.ApplicationInsights

ericmaino committed:
PACKAGE: Upgrading to Newtonsoft 11.0.2

ViniciusSouza committed:
FEATURE: Check for read permission

ViniciusSouza committed:
REFACTOR: Separate Service from Response calls

ericmaino committed:
FEATURE: Add ApplicationVersion and ServiceName

ViniciusSouza committed:
FEATURE: Translate Exceptions into Responses

ViniciusSouza committed:
REFACTOR: Extract IArribaQueryService

ViniciusSouza committed:
TEST: Refactoring ArribaManagementServiceTests

Creating a ArribaServiceBase to be used as a base class for Management and Query services

ericmaino committed:
FEATURE: Naive App Insights Implementation

ViniciusSouza committed:
REFACTOR: Decouple ArribaQueryServices from ArribaQueryApplication

ViniciusSouza committed:
FEATURE: Use ParamChecker guard methods

ViniciusSouza committed:
FEATURE: Add OAuthExtension

ViniciusSouza committed:
REFACTOR: Use Extension OAuthExtension

ViniciusSouza committed:
REFACTOR: Split out host initalization

ericmaino committed:
FIX: Normalize api endpoints using versioning

ericmaino committed:
Merge pull request #15 from ViniciusSouza/visouza/main-ArribaServerRefactor

FEATURE: Add support for AspNet Controller

ericmaino committed:
REFACTOR: Separate Partition Tests

ericmaino committed:
REFACTOR: Extract PartitionConvert and add tests for Ubuntu issue

ericmaino committed:
BUILD: Remove duplicate build and add common parameters

ericmaino committed:
REFACTOR: Extract VerifyNoColumnsAreMissing

ericmaino committed:
REFACTOR: Rename values : datablock

ericmaino committed:
REFACTOR: Improve Partition sorting readability

ericmaino committed:
REFACTOR: Improve partition insert readability

ericmaino committed:
FIX: Ensure partitions are inserted in order

ericmaino committed:
Merge pull request #17 from ericmaino/ericmai/adapter-refactor

REFACTOR: Structures code to better support ports and adapters pattern

ViniciusSouza committed:
TEST: Unit test for ArribaQueryServices

ericmaino committed:
Merge pull request #18 FEATURE: Enable Basic Application Insights

FEATURE: Enable Basic Application Insights

ericmaino committed:
PROJECT: Removing old unused solutions

ViniciusSouza committed:
REFACTOR: Consolidating contract IArribaAuthorization

ViniciusSouza committed:
REFACTOR: Expand contract IArribaServerConfiguration

ViniciusSouza committed:
FIX: Use IArribaServerConfiguration

ViniciusSouza committed:
FIX: Remove unused property _claimsAuth

ViniciusSouza committed:
REFACTOR: Rename namespace to Arriba.Server.Authorization

ericmaino committed:
PACKAGE: Adding DependencyInjection

ViniciusSouza committed:
FIX: Initializing container at constructor

Without having the composition container initialized it is not possible
to add types to it

ViniciusSouza committed:
FIX: Throwing exception when in Developer mode

ViniciusSouza committed:
REFACTOR: Extract security properties to ISecurityConfiguration

ViniciusSouza committed:
FIX: Use one instance of composition host

ViniciusSouza committed:
FEATURE: Add ArribaAuthorizationGrantDecorator

Overrides the returned permission when Authentication is not enabled.

ericmaino committed:
REFACTOR: Unify on the AspNET Composition Model

ericmaino committed:
PACKAGE: Remove Composition Dependencies

ericmaino committed:
Merge pull request #21 from authorization-decorator

Merge pull request #21 from ViniciusSouza/visouza/authorization-refactor

ericmaino committed:
REFACTOR: Composition to use explcit defiintions

ericmaino committed:
Merge pull request #22 REFACTOR: Unify on ASP.NET Composition Model

REFACTOR: Unify on ASP.NET Composition Model

ViniciusSouza committed:
FIX: Initializing ArribaServices

ViniciusSouza committed:
REFACTOR: Add ITelemetry to IArribaManagementService

ViniciusSouza committed:
FIX: Using Newtonsoft.Json as the default serializer

By default the AspNet.Core Serializer on 3.0+ is the System.Text.Json, the current implementation of the library doesn't support Fields only Properties even public fields.

There is a change that will be part of .net 5.0 that will provide a option to change this behaviour
https://github.com/dotnet/runtime/pull/36986

kjeffc committed:
WIP: Initial OpenTelemetry addition

kjeffc committed:
WIP: Adding Activity to suggest method

kjeffc committed:
BUG: WorkItemCrawler and Suggest method broken

kjeffc committed:
Reverting appSettings to CSE settings

mattklup commented on src/Arriba/Arriba.Server/Program.cs:
is tracerProvider being used directly or does it need to be alive?

mattklup commented on src/Arriba/Arriba.Server/Program.cs:
Probably Run

kjeffc commented on src/Arriba/Arriba.Server/Program.cs:
I'm not sure, I think it stays alive.

Here: https://opentelemetry.io/docs/, under the "Configure the OpenTelemetry API" section it says that it acts as the namespace for all the spans under this service

kjeffc commented on src/Arriba/Arriba.Server/Program.cs:
Sorry can you clarify what this means

ViniciusSouza committed:
REFACTOR: Separate Monitor contract to ITelemetry

ViniciusSouza committed:
FEATURE: Add ArribaQueryController

ViniciusSouza committed:
FEATURE: Support for ArribaRoutes V1

Some Rest API routes was not well formed, in order to move forward to V2 without breaking the UI
was added a ArribaActionController that demanded a Refactor on Arriba's controllers

mattklup commented on src/Arriba/Arriba.Server/Program.cs:
sorry, i was responding to your comment.  it should likely wrap around the Run() call

mattklup commented on src/Arriba/Arriba.Server/Program.cs:
I am curious how that namespace state is managed.  some sort of global/static state?
